library/Icingadb/ProvidedHook/Notifications/V1/Source.php

```diff
@@ -0,0 +1,200 @@
+    {
+        return 'icinga2';
+    }
+
+    public function getSourceLabel(): string
+    {
+        return 'Icinga';
+    }
+
+    public function getSourceIcon(): Icon
+    {
+        return new IcingaIcon('icinga');
+    }
+
+    public function getRuleFilterTargets(int $sourceId): array
+    {
+        return [
+            'host' => $this->translate('Hosts only'),
+            'service' => $this->translate('Services only'),
+            self::TYPE_ALL => $this->translate('Hosts and Services')
+        ];
+    }
+
+    public function getRuleFilterEditor(string $filter): SearchEditor
+    {
+        if ($filter === 'host' || $filter === 'service' || $filter === self::TYPE_ALL) {
+            $type = $filter;
+            $filter = '';
+        } else {
+            try {
+                $data = json_decode($filter, true, flags: JSON_THROW_ON_ERROR);
+            } catch (JsonException $e) {
+                Logger::error('Failed to parse rule filter configuration: %s (Error: %s)', $filter, $e);
+                throw new ConfigurationError($this->translate(
+                    'Failed to parse rule filter configuration. Please contact your system administrator.'
+                ));
+            }
+
+            if ($data['version'] !== 1 || ! isset($data['config']['type']) || ! isset($data['config']['filter'])) {
+                Logger::error('Invalid rule filter configuration: %s', $filter);
+                throw new ConfigurationError($this->translate(
+                    'Invalid rule filter configuration. Please contact your system administrator.'
+                ));
+            }
+
+            $type = $data['config']['type'];
+            $filter = $data['config']['filter'];
+        }
+
+        $editor = new SearchEditor();
+        $editor->setQueryString($filter);
+        $editor->setSuggestionUrl(Url::fromPath(
+            'icingadb/suggest/restriction-column',
+            ['_disableLayout' => true, 'showCompact' => true, 'type' => $type]
+        ));
+        $editor->getParser()->on(QueryString::ON_CONDITION, function (Condition $condition) {
+            if ($condition->getColumn()) {
+                if (array_key_exists($condition->getColumn(), $this->allowedColumns)) {
+                    $condition->metaData()->set('columnLabel', $this->allowedColumns[$condition->getColumn()]);
+                } elseif (preg_match('/^(host|service)\.vars\.(.*)/i', $condition->getColumn(), $m)) {
+                    $prefix = $m[1] === 'host' ? $this->translate('Host') : $this->translate('Service');
+                    $condition->metaData()->set('columnLabel', $prefix . ' ' . $m[2]);
+                }
+            }
+        });
+        $editor->on(SearchEditor::ON_VALIDATE_COLUMN, function (Condition $condition) {
+            if (! array_key_exists($condition->getColumn(), $this->allowedColumns)) {
+                if (! preg_match('/^(?:host|service)\.vars\./i', $condition->getColumn())) {
```

Comment:
These conditions can be merged.

Reply:
Only by breaking the line length limit, which leads to breaking the conditions onto their own lines again, not enhancing readability at all.